### PR TITLE
fix: remove JsonElement type, rewrite variable extraction

### PR DESCRIPTION
**Pull request by fenn7**
🦀

### CODE REVIEW

**Comment by williambl** [comment]
I want to look at this before merge

**Comment by fenn7** [comment]
do not merge until will has looked at this

**Comment by williambl** [comment]
conflicts need fixing before merge

**Review by williambl — CHANGES_REQUESTED** [review]
Keep the datatype datatype, but I think maybe for the extract variable node, the setting should be of the 'ReturnType' enum - then, on the front-end, we can have a dropdown that only shows those values.

**Comment by fenn7** [comment]
see attached

**Review by ShaksterNano — APPROVED** [review]

**Comment by fenn7** [comment]
@Will BL
changed list return so it works with json objects

**Comment by will** [comment]
> @Will BL
> 
> changed list return so it works with json objects

Cool, but can we do XML too?

File: basic_nodes/src/main/java/gay/oss/gatos/basicnodes/VariableExtractionNodeType.java [hunk 1]
```diff
@@ -28,23 +40,93 @@ public Set<NodeConnector.Input<?>> inputs(UUID nodeId, Map<String, DataBox<?>> s
...
+    }
+
+    private CompletableFuture handleListReturn(DataType dataType, JsonElement jsonElement) {
+        List<Object> outputList = new ArrayList<>();
+        if (jsonElement instanceof JsonArray arr && arr.size() > 0) {
+            arr.forEach(json -> outputList.add(this.jsonToReturnableType(json)));
+        } else {
+            outputList.add(this.jsonToReturnableType(jsonElement));
+        }
+        return CompletableFuture.completedFuture(dataType.optionalOf().create(
```
Comment (williambl) [inline basic_nodes/src/main/java/gay/oss/gatos/basicnodes/VariableExtractionNodeType.java:85]: No need to return optionals for list types - just return empty list if the variable doesn't exist. We don't want users to have to sometimes unwrap things _twice_.

**Review by williambl — CHANGES_REQUESTED** [review]

File: basic_nodes/src/main/java/gay/oss/gatos/basicnodes/VariableExtractionNodeType.java (hunk 1 above)
Comment (fenn7) [reply basic_nodes/src/main/java/gay/oss/gatos/basicnodes/VariableExtractionNodeType.java]: see updated

**Comment by williambl** [comment]
Will review again once this is up-to-date with master. We should wait until #45 is merged for it.

**Review by williambl — APPROVED** [review]

**Pull request merged**